src/sync.h

```diff
@@ -328,10 +328,10 @@ class SCOPED_LOCKABLE CMutexReadLock
 {
 private:
     boost::shared_lock<Mutex> lock;
-    uint64_t lockedTime;
-    const char *name;
-    const char *file;
-    int line;
+    uint64_t lockedTime = 0;
+    const char *name = "unknown-name";
+    const char *file = "unknown-file";
+    int line = 0;
```

Comment:
Good point. I don't think any lock is created anywhere with a negative line number, but I'll check just to be sure. 

However, whether it could become negative also depends on the behavior of expanding `__LINE__` and thus potentially implementation details of the compiler's preprocessor.

Are we sure compilers will never produce a negative value for `__LINE__`, for example in the hypothetical scenario of compiling `stdin`?

I just tested both `gcc`'s `cpp` as well as `clang`'s `cpp` and they seem to start at line no 1 from `stdin`, as expected.

There's also no arithmetic done with the line numbers so I guess we should be safe. I'll update the code.

Reply:
great

Reply:
@sickpig: There's a lot of functions that take line numbers as an `int` in `sync.h`. Should I change all of them to `unsigned`?

Reply:
sure, makes sense to me